src/strategies/samurailegends-generals-balance/index.ts

```diff
@@ -18,21 +18,27 @@ export async function strategy(
   const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
 
   const multi = new Multicaller(network, provider, abi, { blockTag });
-  addresses.forEach((address) =>
-    multi.call(address, options.address, 'erc721GetAllTokensOfOwner', [
-      options.nftAddress,
-      address
-    ])
+
+  multi.call('batch.owners', options.batchAddress, 'erc721BatchOwnerOf', [
+    options.nftAddress,
+    '0',
+    options.treshold.toString()
+  ]);
+  const result = await multi.execute();
+
+  const balances = result.batch.owners.reduce(
+    (prev: Record<string, number>, curr: string) => {
+      if (curr in prev) prev[curr] += 1;
+      else prev[curr] = 1;
+      return prev;
+    },
+    {}
   );
-  const result: Record<string, number[]> = await multi.execute();
 
   return Object.fromEntries(
-    Object.entries(result).map(([address, nftBalance]) => {
-      const balance = nftBalance.reduce((prev, curr) => {
-        if (curr >= 0 && curr < 5000) return prev + 1;
-        return prev;
-      }, 0);
-      return [address, balance];
+    addresses.map((address: string) => {
+      const balance = balances[address] || 0;
+      return [address, balance * options.multiplier];
```

Comment:
```suggestion
      return [address, balance * (options.multiplier ? options.multiplier : 1)];
```

Reply:
Merging this PR without this, as there are no open/closed proposals with this strategy :)